onnxruntime/test/providers/cpu/nn/pool_fp16_op_test.cc

```diff
@@ -3,7 +3,7 @@
 
 #include "core/mlas/inc/mlas.h"
 
-#if defined(MLAS_F16VEC_INTRINSICS_SUPPORTED) || defined(COREML_ENABLE_MLPROGRAM)
+#if defined(MLAS_F16VEC_INTRINSICS_SUPPORTED) || defined(COREML_ENABLE_MLPROGRAM) || defined(USE_XNNPACK)
```

Comment:
```suggestion
#if defined(MLAS_F16VEC_INTRINSICS_SUPPORTED) || defined(COREML_ENABLE_MLPROGRAM) || defined(XNNPACK_FP16_SUPPORTED)
```

Sorry - I mislead you and using USE_XNNPACK is not sufficient in this case given there are limitations on when the xnnpack fp16 kernels are available that require XNNPACK_FP16_SUPPORTED to be used.